provider/cluster_rosa_classic_resource.go

```diff
@@ -1326,47 +1340,55 @@ func (r *ClusterRosaClassicResource) upgradeClusterIfNeeded(ctx context.Context,
 	if err != nil {
 		return fmt.Errorf("failed to parse desired cluster version: %v", err)
 	}
-	if currentVersion.GreaterThanOrEqual(desiredVersion) {
+	if currentVersion.GreaterThan(desiredVersion) {
 		tflog.Debug(ctx, "No cluster version upgrade needed.")
+		if requestedVersionChanged {
+			// User changed the version they want, but actual is higher. We
+			// don't support downgrades.
+			return fmt.Errorf("cluster version is already above the requested version")
+		}
 		return nil
 	}
+	cancelingUpgradeOnly := desiredVersion.Equal(currentVersion)
```

Comment:
correct me if I'm wrong, but it seems that if cancelingUpgradeOnly is true - you can return from that function